src/App.tsx

```diff
@@ -1,14 +1,55 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [isLoader, setIsLoader] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [methodFilter, setMethodFilter] = useState('all');
+  const [value, setValue] = useState('');
+
+  const [activeTodo, setActiveTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setIsLoader(true);
+
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setFilteredTodos(todosFromServer);
+      })
+      .finally(() => setIsLoader(false));
+  }, []);
+
+  useEffect(() => {
+    let filtered = todos;
+
+    if (methodFilter === 'active') {
+      filtered = todos.filter(todo => !todo.completed);
+    }
+
+    if (methodFilter === 'completed') {
+      filtered = todos.filter(todo => todo.completed);
+    }
```

Comment:
Use a switch case instead of if conditions.
Also, create a enum for 'all', 'active', 'completed' and use it everywhere